packages/react-instantsearch/src/components/Hits.js

```diff
@@ -1,39 +1,32 @@
+import React from 'react';
 import PropTypes from 'prop-types';
-import React, { Component } from 'react';
 import classNames from './classNames.js';
 
 const cx = classNames('Hits');
 
-class Hits extends Component {
-  render() {
-    const { hitComponent: ItemComponent, hits } = this.props;
-    return (
-      <div {...cx('root')}>
-        {hits.map(hit => <ItemComponent key={hit.objectID} hit={hit} />)}
-      </div>
-    );
-  }
-}
+const Hits = ({ hits, hitComponent: HitComponent }) => (
+  <div {...cx('root')}>
+    {hits.map(hit => <HitComponent key={hit.objectID} hit={hit} />)}
+  </div>
+);
 
 Hits.propTypes = {
   hits: PropTypes.array,
   hitComponent: PropTypes.func.isRequired,
 };
 
-/* eslint-disable react/display-name */
 Hits.defaultProps = {
-  hitComponent: hit => (
+  hitComponent: props => (
```

Comment:
It's more the `hit` which is passed into the single hit, not one of the props of Hits, right?

Reply:
Actually `hit` is one of the props pass to the `HitComponent`. Maybe destruct the props will be easier to understand:

```js
Hits.defaultProps = {
  hitComponent: ({ hit }) => (
    ...
  ),
}
```

Reply:
yes, what we really want to display is the destructured, so we should stringify that 👍

Reply:
At the end maybe it's better to stringify all the props pass to the `HitComponent`. When you render the `Hits` component without the custom component we want to see all the props passed to it. Can be confusing to stringify only the `hit`, the user may think that the object displayed are the props. In fact it's only the `hit` property on the props. WDYT?

> https://www.argos-ci.com/algolia/react-instantsearch/builds/15116

Reply:
sure, makes sense. Personally I'd like it more if we just passed the attributes as props instead of the hit, but that would be a breaking change. Thanks for taking a look at this. Seems like keeping it non-destructured is the good call here indeed

Reply:
Completely agree! I will create an issue as a reminder for a later release with BC. Thanks!